lib/AbstractHandle/Utils/Handler.py

```diff
@@ -241,8 +241,8 @@ def add_read_acl(self, hids, token, username=None):
 
             node_id = handle.get('id')
             try:
-                self.shock_util.add_read_acl(node_id, token, username=username)
-            except Exception:
-                raise ValueError("Unable to set acl(s) on handles {}".format(handle.get('hid')))
-
+                self.shock_util.add_read_acl(node_id, username=username)
+            except Exception as e:
+                raise ValueError("Unable to set acl(s) on handles {}\n{}".format(handle.get('hid'),
+                                                                                 e))
```

Comment:
Does this add the stacktrace into the message or just the exception type and message?

Regardless, I think you should add `from e` so the stacktrace for the primary exception is appended to the new exception.

Reply:
👍  cool. Thanks.